src/providers/list/utils/updateInventoryIfChanged.ts

```diff
@@ -0,0 +1,59 @@
+import type { ProviderInventory, ProvidersInventoryList } from '@kubev2v/types';
+
+import { hasObjectChangedInGivenFields } from '../../../modules/Providers/utils/helpers/hasObjectChangedInGivenFields';
+
+import { PROVIDER_TYPES } from './constants';
+
+export const updateInventoryIfChanged = (
+  newInventoryList: ProvidersInventoryList,
+  setInventory: (value: React.SetStateAction<ProvidersInventoryList | null>) => void,
+  oldDataRef: React.MutableRefObject<ProvidersInventoryList | null>,
+  setLoading: (value: React.SetStateAction<boolean>) => void,
+  fieldsToAvoidComparing: string[],
+  // eslint-disable-next-line @typescript-eslint/max-params
```

Comment:
instead of adding comment , split the function

Reply:
this function is 50 lines of code, can be split into several functions for reliability and maitnablity